python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -857,17 +775,141 @@ class _FunctionProxy(_CallableProxyMixin):
 
     __name__: str
 
-    def __init__(self, fast: Callable | _Unusable, slow: Callable):
+    def __init__(
+        self,
+        fast: Callable | _Unusable,
+        slow: Callable,
+        *,
+        assigned=None,
+        updated=None,
+    ):
         self._fsproxy_fast = fast
         self._fsproxy_slow = slow
-        functools.update_wrapper(self, slow)
+        assigned = (
```

Comment:
These just reassign None when the value is None, but in a slightly nonobvious way. Let’s simplify to
```python
if assigned is None:
    assigned = functools.WRAPPER_ASSIGNMENTS
```
and similarly for `updated`.